### PR TITLE
fix loop error, add unit test

### PR DESCRIPTION
**Pull request by kovach**
incorrect loop update

### CODE REVIEW

File: src/sats_management.c [hunk 1]
```diff
@@ -94,16 +94,17 @@ void set_reference_sat_of_prns(const u8 ref_prn, const u8 num_sats, u8 *prns)
     memcpy(old_prns, prns, num_sats * sizeof(u8));
     u8 set_old_yet = 0;
     prns[0] = ref_prn;
-    for (u8 i=1; i<num_sats; ) {
+    for (u8 i=1; i<num_sats; i++) {
```
Comment (henryhallam) [inline src/sats_management.c]: lol, that'd do it

File: tests/check_sats_management.c [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  sats_management_t sats_management;
+  sats_management.num_sats = 4;
+  sats_management.prns[0] = 2;
+  sats_management.prns[1] = 1;
+  sats_management.prns[2] = 3;
+  sats_management.prns[3] = 4;
+
+  set_reference_sat_of_prns(new_ref, num_sats, prns);
+  /* Just check the sats_management prns update */
+  set_reference_sat(new_ref, &sats_management, 0, 0, 0);
```
Comment (henryhallam) [inline tests/check_sats_management.c]: Does this actually test anything? relying on the assert in set_reference_sat()?

**Comment by henryhallam** [comment]
I'm kind of surprised this wasn't seen more often in field testing than it was.

File: tests/check_sats_management.c (hunk 2 above)
Comment (kovach) [inline tests/check_sats_management.c]: it tests that they return

Comment (henryhallam) [inline tests/check_sats_management.c]: Lol

**Comment by cbeighley** [comment]
lgtm. Can you merge and then update the associated piksi_firmware PR

**Pull request merged**